### ISSUE TITLE
If ORCID null, don't include in datapackage

### ISSUE THREAD

**Issue by e-belfer** [catalyst-cooperative/pudl#3005]
### Describe the bug

Datapackages in the `pudl-archiver` repository are generated using methods imported from the `pudl` repository.

* *The problem:* ORCID IDs in `datapackage.json` files for every dataset where contributors don't have an ORCID ID are appearing as `orcid: null`.
* *Where is this problem taking place?* `orcid` is a field of the `Contributor` class in `pudl.metadata.classes.py`, and can either be a string or a `None` type. For most contributors, this field is not provided in the `CONTRIBUTORS` dictionary in `pudl.metadata.sources.py`. When we initialize the `Contributor` class using the `Contributor.from_id()` method in `pudl.metadata.classes.py`, it produces a dictionary with a `null` value. 
* *Where are datapackages being generated?* The method used to create a new datapackage in the `pudl-archiver` repository (in `pudl_archiver.frictionless.py`), is `DataSource.from_id()` in `pudl.metadata.classes.py`. The `Contributor` class is also called directly in the `pudl-archiver` repository, so the fix will need to be made in both the Contributor and the `DataSource` classes.

### Bug Severity
How badly is this bug affecting you?
 - **Medium:** With some effort, I can work around the bug.
This occasionally causes issues but is also the status quo in all existing archives.

### To Reproduce
See the `datapackage.json` file for any existing data archive, e.g.: https://zenodo.org/records/8164776

To produce a datapackage from an existing dataset, install PUDL locally [following these instructions](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/dev_setup.html) and run:

```py
from pudl.metadata.classes import DataSource
DataSource.from_id("eia860").contributors
```
You can replace "eia860" with any of the data source IDs in `pudl.metadata.sources.py`.

### Expected behavior
The `orcid` field should only be included as a field in the datapackage when it exists.

### Software Environment?
 - Operating System. (e.g. MacOS 14.5, Ubuntu 22.04, Windows Subsystem for Linux v2)
 Ubuntu 22.04

 - Python version and distribution (e.g. Anaconda Python 3.10.6)
Python 3.11.6 

 - How did you install PUDL?
 `git clone dev`

**Comment by yolandazzz13** [comment]
Hi, I'm a student at University of Michigan who's currently working on a final project in a software engineering course that expect us to make contributions to open source github projects. I wonder if you can assign me to this issue? @catalyst-cooperative/com-dev Also , it would be awesome if I can be told how to generate new datapackage.json files. Thank you in advance!

**Comment by e-belfer** [comment]
@yolandazzz13 Awesome, happy you found us! Let me flesh this issue out a bit more to make it clearer what needs to get done and make sure it's a good fit for a first-time contributor.

**Comment by yolandazzz13** [comment]
Hi,

Thank you so much for the detailed description. I was able to quickly
locate the issue and figure out a solution for it. I also wanted to let you
know that I was able to fix another issue #3592 during the past week. And
I've submitted a PR for you and the developing team to review. I'd
really appreciate it if you could take some time to approve it. In fact, I
would be able to receive some extra credit for the software engineering
course that I mentioned before if my PR was accepted before December.
Thanks again for your support!

Best,
Yolanda Zhou
